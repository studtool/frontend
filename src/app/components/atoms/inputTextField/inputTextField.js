import styled from 'styled-components';

export const InputTextField = styled.input``;
