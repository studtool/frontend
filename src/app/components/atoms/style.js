export const colors = {
    primary : '#5952de',
    primaryLight: '#917fff',
    primaryDark: '#0928ab',
    secondary: '#f44336',
    secondaryLight: '#ff7961',
    secondaryDark: '#ba000d',
    textOnPrimary: '#ffffff',
    textOnSecondary: '#000000',
    black: '#000000',
    gray: '#757575',
    white: '#ffffff',
}
