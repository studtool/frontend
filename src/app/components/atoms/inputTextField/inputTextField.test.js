import React from 'react';
import {InputTextField} from './inputTextField';
import {colors} from '../style';
import renderer from 'react-test-renderer'
import 'jest-styled-components'

test('inputTextField test', () => {
    const tree = renderer.create(<InputTextField/>).toJSON();

    expect(tree).toHaveStyleRule('color', colors.brlack);
    expect(tree).toHaveStyleRule('background', colors.white);
    expect(tree).toHaveStyleRule('border', 'none');
    expect(tree).toHaveStyleRule('border-bottom', '1px solid '+colors.gray);
    expect(tree).toHaveStyleRule('border-bottom', '2px solid '+colors.black, {
        modifier: ':hover',
    });
    expect(tree).toHaveStyleRule('border-bottom', '2px solid '+colors.primaryDark, {
        modifier: ':focus',
    });

    expect(tree).toHaveStyleRule('outline', 'none');

    expect(tree).toMatchSnapshot();
});
