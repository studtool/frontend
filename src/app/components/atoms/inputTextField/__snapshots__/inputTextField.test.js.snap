// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inputTextField test 1`] = `
.c0 {
  background: #ffffff;
  border: none;
  border-bottom: 1px solid #757575;
  outline: none;
}

.c0:hover {
  border-bottom: 2px solid #000000;
}

.c0:focus {
  border-bottom: 2px solid #0928ab;
}

<input
  className="c0"
/>
`;
